### PR TITLE
feat(@desktop/wallet): Use new activity model for HistoryView

### LINKED ISSUE #11019: Use new activity model in QML HistoryView

### PR DESCRIPTION
**Pull request by Cuteivist**
closes #11019

**Multitransactions will be handled in next issue: #11071** 

### What does the PR do

* Uses new activity model
* Handles filtering activity model depending on UI
* Calls load more on scrolling
* Some small additional fixes

### Affected areas

* Wallet
* NFT - currently the data is missing in the model

### Screenshot of functionality (including design for comparison)

https://github.com/status-im/status-desktop/assets/11396062/c3b486f4-c38f-4b6c-bd3a-bc0e7aed08a8

### CODE REVIEW

File: ui/imports/shared/views/HistoryView.qml [hunk 1]
```diff
@@ -112,14 +104,15 @@ ColumnLayout {
 
             // LocaleUtils is not accessable from inside expression, but local function works
             property var formatDate: (ms) => LocaleUtils.formatDate(ms, Locale.ShortFormat)
-            sorters: RoleSorter {
-                roleName: "timestamp"
-                sortOrder: Qt.DescendingOrder
+            sorters: ExpressionSorter {
```
Comment (stefandunca) [inline ui/imports/shared/views/HistoryView.qml:107]: The sorting is done in SQL query, not at runtime. Now descending by timestamp is activated by default. We have to remove runtime sorting, which is redundant.

**Review by stefandunca — APPROVED** [review]
Good job. Left a comment inline.

File: src/app/modules/main/wallet_section/activity/entry.nim [hunk 2]
```diff
@@ -37,12 +38,18 @@ QtObject:
...
     result.setup()
 
-  proc newTransactionActivityEntry*(tr: ref Item, metadata: backend.ActivityEntry): ActivityEntry =
+  proc newTransactionActivityEntry*(tr: ref Item, metadata: backend.ActivityEntry, fromAddresses: seq[string]): ActivityEntry =
     new(result, delete)
     result.multi_transaction = nil
     result.transaction = tr
     result.isPending = metadata.payloadType == backend.PayloadType.PendingTransaction
     result.metadata = metadata
+    result.activityType = backend.ActivityType.Send
```
Comment (dlipicar) [inline src/app/modules/main/wallet_section/activity/entry.nim:47]: @stefandunca isn't this supposed to come from the backend?

Comment (stefandunca) [reply src/app/modules/main/wallet_section/activity/entry.nim:47]: Yes but didn't want to current work to be blocked by the activity metadata plan, this TODO covers it: https://github.com/status-im/status-desktop/blob/951fa8b0f4d145c5825fa35bdd406360498b739b/src/backend/activity.nim#L147

In the backend, it is already set. I was planning in my current PR to fix this, will replace it when I make the change or @Cuteivist, if you want you can fix it yourself.

**Review by dlipicar — APPROVED** [review]

**Review by alaibe — APPROVED** [review]

File: src/app/modules/main/wallet_section/activity/entry.nim (hunk 2 above)
Comment (stefandunca) [reply src/app/modules/main/wallet_section/activity/entry.nim:47]: @Cuteivist, thanks for the heads up, I will take a look later on

**Review by anastasiyaig — APPROVED** [review]

**Pull request merged**